OpenRA.Mods.Common/UtilityCommands/ImportLegacyMapCommand.cs

```diff
@@ -9,33 +9,344 @@
+				.Where(kv => Exts.ParseIntegerInvariant(kv.Value) > 0)
+				.Select(kv => Pair.New(Exts.ParseIntegerInvariant(kv.Key),
+					LocationFromMapOffset(Exts.ParseIntegerInvariant(kv.Value), mapSize)));
+
+			// Add waypoint actors
+			foreach (var kv in wps)
+			{
+				if (kv.First <= 7)
+				{
+					var ar = new ActorReference("mpspawn")
+					{
+						new LocationInit((CPos)kv.Second),
+						new OwnerInit("Neutral")
+					};
+
+					map.ActorDefinitions.Add(new MiniYamlNode("Actor" + actorCount++, ar.Save()));
+				}
+				else
+				{
+					var ar = new ActorReference("waypoint")
+					{
+						new LocationInit((CPos)kv.Second),
+						new OwnerInit("Neutral")
+					};
+
+					map.ActorDefinitions.Add(new MiniYamlNode("waypoint" + kv.First, ar.Save()));
+				}
+			}
+		}
+
+		static void LoadSmudges(IniFile file, string section, int mapSize, Map map)
+		{
+			foreach (var s in file.GetSection(section, true))
+			{
+				// loc=type,loc,depth
+				var parts = s.Value.Split(',');
+				var loc = Exts.ParseIntegerInvariant(parts[1]);
+				var key = "{0} {1},{2} {3}".F(parts[0].ToLowerInvariant(), loc % mapSize, loc / mapSize, Exts.ParseIntegerInvariant(parts[2]));
+				map.SmudgeDefinitions.Add(new MiniYamlNode(key, ""));
+			}
+		}
+
+		// TODO: fix this -- will have bitrotted pretty badly.
+		static Dictionary<string, HSLColor> namedColorMapping = new Dictionary<string, HSLColor>()
+		{
+			{ "gold", HSLColor.FromRGB(246, 214, 121) },
+			{ "blue", HSLColor.FromRGB(226, 230, 246) },
+			{ "red", HSLColor.FromRGB(255, 20, 0) },
+			{ "neutral", HSLColor.FromRGB(238, 238, 238) },
+			{ "orange", HSLColor.FromRGB(255, 230, 149) },
+			{ "teal", HSLColor.FromRGB(93, 194, 165) },
+			{ "salmon", HSLColor.FromRGB(210, 153, 125) },
+			{ "green", HSLColor.FromRGB(160, 240, 140) },
+			{ "white", HSLColor.FromRGB(255, 255, 255) },
+			{ "black", HSLColor.FromRGB(80, 80, 80) },
+		};
+
+		public static void SetMapPlayers(string section, string faction, string color, IniFile file, List<string> players, MapPlayers mapPlayers)
+		{
+			var pr = new PlayerReference
+			{
+				Name = section,
+				OwnsWorld = section == "Neutral",
+				NonCombatant = section == "Neutral",
+				Faction = faction,
+				Color = namedColorMapping[color]
+			};
+
+			var neutral = new[] { "Neutral" };
+			foreach (var s in file.GetSection(section, true))
+			{
+				switch (s.Key)
+				{
+				case "Allies":
+					pr.Allies = s.Value.Split(',').Intersect(players).Except(neutral).ToArray();
+					pr.Enemies = s.Value.Split(',').SymmetricDifference(players).Except(neutral).ToArray();
+					break;
+				default:
+					Console.WriteLine("Ignoring unknown {0}={1} for player {2}", s.Key, s.Value, pr.Name);
+					break;
+				}
+			}
+
+			// Overwrite default player definitions if needed
+			if (!mapPlayers.Players.ContainsKey(section))
+				mapPlayers.Players.Add(section, pr);
+			else
+				mapPlayers.Players[section] = pr;
+		}
+
+		public static string[] OverlayActors = new string[]
```

Comment:
These are (at least in principle) mod specific, so should be hoisted out to the RA/TD importers.